src/dsp_tools/utils/rosetta.py

```diff
@@ -0,0 +1,67 @@
+from pathlib import Path
+import shutil
+import subprocess
+
+from dsp_tools.models.exceptions import UserError
+from dsp_tools.utils.project_create import create_project
+from dsp_tools.utils.xml_upload import xml_upload
+
+def upload_rosetta() -> bool:
```

Comment:
this method does multiple things... could be split up into something like: 

```python
def upload_rosetta() -> bool:
    if _require_clone():
        _clone_rosetta()
    else:
        _pull_rosetta()
    _upload_rosetta()
```

then it remains in one abstraction level (here, the highest), and can actually be read like natural language.